Allow opting out of forced OIDC email verification (#15847)
gravitational/teleport: docs/pages/access-controls/sso/oidc.mdx

A list of available optional prompt parameters are available from the
[OpenID website](https://openid.net/specs/openid-connect-core-1\_0.html#AuthRequest).

> ### Optional: Disable Email Verification

Reviewer: ```suggestion
### Optional: Disable email verification
```
We use sentence case in section headings

> By default, Teleport will validate the `email_verified` claim, and users who attempt to sign in without a verified email address will be prevented from doing so.

Reviewer 2: ```suggestion
By default, Teleport will validate the `email_verified` claim and reject login attempts from users without a verified email address.
```

This is more succinct but it's up to you to decide if it actually sounds better.

> For testing and other purposes, you may opt out of this behavior by enabling `allow_unverified_email` in your OIDC Connector. This option weakens the overall security of the system, so we do not recommend enabling it.

Reviewer: ```suggestion
For testing and other purposes, you may opt out of this behavior by enabling `allow_unverified_email` in your OIDC connector. This option weakens the overall security of the system, so we do not recommend enabling it.
```

A list of available optional prompt parameters are available from the
[OpenID website](https://openid.net/specs/openid-connect-core-1\_0.html#AuthRequest).

### Optional: Disable Email Verification

> By default, Teleport will reject OIDC claims/users that do not have a verified email address.
> This information is given to Teleport via the `email_verified` OIDC claim.
> 
> For testing and other purposes, this may be disabled as shown below. Note that this may harm
> security and extreme care should be taken outside of testing environments.

Reviewer 3: ```suggestion
By default, Teleport will validate the `email_verified` claim, and users who attempt to sign in without a verified email address.

For testing and other purposes, you may opt out of this behavior by enabling `allow_unverified_email` in your OIDC Connector. This option weakens the overall security of the system, so we do not recommend enabling it.
```
